Repository browser

GrantWWalker opened the pull request:
Created a new general lf-repository-browser component
Defined the LfTreeNodeService to be used with the repository browser to provide the data

Created a new lf-selection-list component to help with listings that need keyboard accessibility

Created the Selectable class to handle selection within a list

GrantWWalker committed:
Issue #53 - Created the base for the repository browser.
Created the ILFRepositoryService which is a more generic version of the lf-tree.service
Created the repository-browser.directive which is a modified version of the flat-tree.directive

GrantWWalker committed:
Merge branch '13.x' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Commit:
Merge branch '13.x' of https://github.com/Laserfiche/lf-ui-components into repository-browser

GrantWWalker committed:
Added the lf-repository-browser-documentation.component which is not currently working

Changed some code with getting data and how the repo-browser is showing it.

GrantWWalker committed:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Commit:
updated repository-browser-documentation to show repository-browser, but not showing any data yet

Commit:
update repository-browser, fix styling of scroll-viewport, add breadcrumbs

Commit:
update folder-browser and file-explorer to use new breadcrumbs component from shared, remove breadcrumbs component in flat-tree-components-module

GrantWWalker committed:
Fixed some styling with the messages
Fixed some overflow and styling issues with the entry listing itself
Added some extra cases to the documentation

Commit:
update repository-browser documentation
test with multiple select

GrantWWalker committed:
Added an example of dynamic data to the documentation

Removed some unused CSS
Renamed old CSS classes to match with the new naming
Added a way for the lf-repository-browser.component know when to pull more data
Fixed some issues about not setting the current entry in the repository-browser.directive which causes issues for the dynamic loading of data

GrantWWalker committed:
Merged changes with the remote branch

Commit:
don't disable non-selectable entries

Commit:
update select behavior and interfaces

Commit:
update repository interfaces to TreeNode interface, add support for explicit paging

Commit:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into alex/update-select-behavior

GrantWWalker committed:
Create tests for the repository-browser.directive
Removed all the lf-repository-browser-documentation.component tests for now so they can be reworked

Added the breadcrumbClicked event to the lf-breadcrumbs.component which will handle all of the breadcrumb changes in the component itself so outside components do not have to

Removed the base-field.directive.spec's describe because it throws a warning when a describe does not have any tests

GrantWWalker committed:
Updated the lf-repository-browser-documentation, ILfTreeNodeService, lf-repository-browser.component and repository-browser.directive to handle the naming/service changes

Updated the repository-browser.directive tests to be more stable and handle the new service changes

Commit:
add ability to select via keyboard

Commit:
Add ability to multiselect,
add keyboard accessibility for multi select
add service for selection into shared

Commit:
Refactor ILfSelectable,
do not expose isSelected property in user TreeNode

Commit:
update interface to not have params,
update setSelectedValues to be Async to account for getting more values before emitting value,
add setTimeout to allow default behavior to occur first (this needs to be updated to be more robust)
removed unused code (i.e. for breadcrumbs)

GrantWWalker committed:
Removed the shouldShowNoMatchesMessage because of the way the filtering happens now

GrantWWalker committed:
Updated the multiple input to be on the repository-browser.directive so that it can be passed to the selectable property

Changed some styling around the selection and hover colors

Changed the way the enter button is handled

GrantWWalker committed:
Created the lf-list.component to take over the job of keyboard accessibility and selection instead of having that built into the repository-browser

Updated the lf-repository-browser.component to use the new lf-list as the internal holder of data.

GrantWWalker committed:
Created the _palette-lf.scss to help describe the general colors that match the Laserfiche style guide
Added some global css variables to allow for usage in some of the components

Removed the multi-selection checkbox for items that are not selectable

GrantWWalker committed:
Fixed linting errors

GrantWWalker committed:
Fixed some broken Custom Component mappings

GrantWWalker committed:
Fixed issue with lf-list.component not looking for the right nodename

GrantWWalker committed:
Updated the lf-list to be the lf-selection-list
Moved the lf-selection-list into its own component project

Merged the repository-browser.directive into the lf-repository-browser.component
Updated the tests for the lf-repository-browser.component

GrantWWalker committed:
Updated the reference to the LfSelectionListModule in the lf-repository-browser.component.spec

GrantWWalker committed:
Fixed an issue with the ctrl + shift clicking behavior
Fixed an issue with functionality of holding shift and using the arrow keys to select items

Created tests for the LfSelectable

GrantWWalker committed:
Fixed linting errors
Removed the fdescribe so all tests will run

Ke-Jin-Laserfiche commented on projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts:
```diff
@@ -0,0 +1,42 @@
+export interface TreeNode {
+    id: string;
```
how about LfTreeNode?

Ke-Jin-Laserfiche commented on projects/ui-components/lf-selection-list/lf-list-option.component.ts:
```diff
@@ -0,0 +1,57 @@
+  templateUrl: './lf-list-option.component.html',
+  styleUrls: ['./lf-list-option.component.css']
+})
+export class LfListOptionComponent  {
```
since this component cannot be accessed by the user, can we mark it internal so the types doesn't go into the types npm package? 
```
/**
 * @internal
 */
@Component({
...
})
```

GrantWWalker replied:
Oh nice is that what @internal is used for? I kept meaning to ask.

alexgomezlf replied:
yeah there's also a stripInternal property in the tsconfig that will strip @internal on compilation but we don't use that, we use it to help our types bundler not include stuff that won't be included in the CDN

GrantWWalker committed:
Updated the @internal markings so they do not show up in the types package

Ke-Jin-Laserfiche commented on projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:
```diff
@@ -0,0 +1,411 @@
+    public localizationService: AppLocalizationService
+  ) {
+    this.scrolledIndexChanged.pipe(debounceTime(200)).subscribe(async () => {
+      if (this._currentEntry == null) {
```
Does this need to be unsubscribed?

Ke-Jin-Laserfiche commented on projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts:
```diff
@@ -0,0 +1,411 @@
+    /** @internal */
+    public localizationService: AppLocalizationService
+  ) {
+    this.scrolledIndexChanged.pipe(debounceTime(200)).subscribe(async () => {
```
Sometimes this get called twice. Is the debounceTime(200) an attempt to fix that behavior? Why is it 200?

GrantWWalker replied:
Sorry I missed that, its really not the best way to fix this but it was sort of arbitrary to just try and catch some of the extra calls if there are any

Ke-Jin-Laserfiche committed:
add maximumChildrenReceived and unsubscribe

Ke-Jin-Laserfiche committed:
add internal to necessary location

Ke-Jin-Laserfiche committed:
Merge pull request #55 from Laserfiche/ke/repository-browser

add maximumChildrenReceived and unsubscribe

GrantWWalker committed:
Added the ILfTreeNodeService to the lf-repository-browser-public-api

Ke-Jin-Laserfiche committed:
set maximumChildrenReceived only when nextpage is undefined

GrantWWalker committed:
Removed the dependency on CoreUtils

GrantWWalker committed:
Added the ability to refresh the repository browser component

GrantWWalker committed:
Fixed linting error

GrantWWalker committed:
Fixed a styling issue with the lf-list-option.component

Commit:
clean up PR
make sure internal is added everywhere needed
add LfRepositoryBrowserModule to cdn exports

GrantWWalker committed:
Allowed for the user to pass the itemSize of their list times
Fixed an issue of focus when the lastFocused index is not currently in the view window for the lf-selection-list.component

GrantWWalker committed:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Ke-Jin-Laserfiche committed:
add lastCalledPage and refreshAsync

Ke-Jin-Laserfiche committed:
expose currentFolder

GrantWWalker committed:
Set isLoading to default to true so that the empty message does not pop up quickly before the component is initialized

GrantWWalker committed:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Ke-Jin-Laserfiche committed:
fix lint

Ke-Jin-Laserfiche committed:
fix lint

Commit:
update documentation for lf-repository-browser

Commit:
fix lint, add semicolon

alexgomezlf commented on projects/ui-components/lf-repository-browser/lf-repository-browser.component.css:
```diff
@@ -0,0 +1,143 @@
+    height: 34px;
+}
+
+::ng-deep .cdk-overlay-pane .mat-menu-panel {
```
will this affect all material styles?

GrantWWalker replied:
It could possibly, but I do not think its needed any more. Let me go through and clean up the styles

GrantWWalker committed:
Removed unused tests

GrantWWalker committed:
Removed unneeded styles

Commit:
PR review updates, delete unnecessary comments

GrantWWalker committed:
Small file cleanup

GrantWWalker committed:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Commit:
update minor version and add Changelog describing changes

GrantWWalker committed:
Added a code snip-it to show the Angular Component

GrantWWalker committed:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

Commit:
change onEntrySelected event to handle CustomeEvent

Commit:
Merge branch 'repository-browser' of https://github.com/Laserfiche/lf-ui-components into repository-browser

GrantWWalker committed:
Updated the styling for the lf-selection-list.component and the implementation of it in the lf-repository-browser.component

Fixed an issue with accidentally pulling data without waiting for the last data pull to finish

GrantWWalker committed:
Reverted the style update for the lf-selection-list.component because of keyboard accessibility issues

Made the lf-breadcrumb a bit more obvious that its clickable

GrantWWalker committed:
Fixed linting error

GrantWWalker committed:
Fixed an issue with when the lastDataCall is errored

Removed some padding from the lf-list-option.component
